store/sqlstore/group_store.go

```diff
@@ -359,6 +359,25 @@ func (s *SqlGroupStore) Delete(groupID string) (*model.Group, error) {
 	return &group, nil
 }
 
+func (s *SqlGroupStore) Undelete(groupID string) (*model.Group, error) {
```

Comment:
The delete and undelete functions share > 98% of the code. Should be have a common parametarized function that sets the delete at value from the param?

Reply:
Thanks for the reviews @harshilsharma63 .

I agree with this, but I think the flow would break 'somewhere' if we use the same function from api->app->store. Since logs would also change and using a boolean in these places would make it less readable.

Plus, (although it'd be very rare) if the delete and undelete functionalities differ by a good amount in future, the refactoring would be difficult. 

If you have a hard opinion on this, I can definitely do the change. Please let me know.

Reply:
Its fine.